data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/BeaconRestApi.java

```diff
@@ -127,7 +127,7 @@
 import tech.pegasys.teku.validator.api.NodeSyncingException;
 import tech.pegasys.teku.validator.coordinator.Eth1DataProvider;
 
-public class BeaconRestApi {
+public class BeaconRestApi implements BeaconRestApiType {
```

Comment:
nit: I'd probably call the interface `BeaconRestApi` and this implementation `ReflectionBasedReaconRestApi` or maybe just `LegacyBeaconRestApi`.  Then the new one would either be `JsonTypeDefinitionBeaconRestApi` or `MigratedBeaconRestApi`.